lib/noble.js

```diff
@@ -1,190 +1,129 @@
 // Pre-load elliptic as a backup
 const elliptic = require('./elliptic')
-const bitcoinerlab = require('@exodus/bitcoinerlab-secp256k1')
-const noble = require('@noble/secp256k1')
+const necc = require('@noble/secp256k1')
+const hmac = require('@noble/hashes/hmac')
+const sha256 = require('@noble/hashes/sha256')
 
-const errors = {
-  EXPECTED_PRIVATE: 'Expected Private',
-  EXPECTED_POINT: 'Expected Point',
-  EXPECTED_SIGNATURE: 'Expected Signature'
+function _interopNamespaceDefault (e) {
+  const n = Object.create(null)
+  if (e) {
+    Object.keys(e).forEach(function (k) {
+      if (k !== 'default') {
+        const d = Object.getOwnPropertyDescriptor(e, k)
+        Object.defineProperty(n, k, d.get ? d : {
+          enumerable: true,
+          get: function () { return e[k] }
+        })
+      }
+    })
+  }
+  n.default = e
+  return Object.freeze(n)
 }
 
+const noble = _interopNamespaceDefault(necc)
+
+// Additional Polyfills as per official noble docs https://github.com/paulmillr/noble-secp256k1?tab=readme-ov-file#usage
+noble.utils.hmacSha256Sync = (key, ...msgs) =>
+  hmac.hmac(sha256.sha256, key, noble.utils.concatBytes(...msgs))
+noble.utils.sha256Sync = (...msgs) => sha256.sha256(noble.utils.concatBytes(...msgs))
+
 function isCompressedPublicKey (array) {
   return array.length === 33
 }
 
-const hexes = Array.from({ length: 256 }, (v, i) => i.toString(16).padStart(2, '0'))
-function bytesToHex (uint8a) {
-  if (!(uint8a instanceof Uint8Array)) { throw new Error('Expected Uint8Array') }
-  let hex = ''
-  for (let i = 0; i < uint8a.length; i++) {
-    hex += hexes[uint8a[i]]
+function hexToNumber (hex) {
+  if (typeof hex !== 'string') {
+    throw new TypeError('hexToNumber: expected string, got ' + typeof hex)
   }
-  return hex
+  return BigInt(`0x${hex}`)
 }
 
-function arrayToBigInt (array) {
-  return BigInt('0x' + bytesToHex(array))
+function normalizeScalar (scalar) {
+  let num
+  if (typeof scalar === 'bigint') {
+    num = scalar
+  } else if (
+    typeof scalar === 'number' &&
+    Number.isSafeInteger(scalar) &&
+    scalar >= 0
+  ) {
+    num = BigInt(scalar)
+  } else if (typeof scalar === 'string') {
+    if (scalar.length !== 64) { throw new Error('Expected 32 bytes of private scalar') }
+    num = hexToNumber(scalar)
+  } else if (scalar instanceof Uint8Array) {
+    if (scalar.length !== 32) { throw new Error('Expected 32 bytes of private scalar') }
+    num = hexToNumber(noble.utils.bytesToHex(scalar))
+  } else {
+    throw new TypeError('Expected valid private scalar')
+  }
+  if (num < 0) throw new Error('Expected private scalar >= 0')
+  return num
+}
+
+function _isPoint (p) {
+  try {
+    return !!noble.Point.fromHex(p)
+  } catch (e) {
+    return false
+  }
+}
+
+function _isSignature (sig) {
+  if (sig?.length < 64) throw new Error('Expected signiture length of 64')
```

Comment:
```suggestion
  if (sig?.length !== 64) throw new Error('Expected signiture length of 64')
```

Reply:
Just remembered the signature length check is already present 
https://github.com/ExodusMovement/secp256k1-node/blob/ap/fast_functions/lib/index.js#L289